addon/plugins/variable-plugin/nodes.ts

```diff
@@ -107,29 +160,44 @@ const emberNodeConfig: EmberNodeConfig = {
           hasRDFaAttribute(node, 'typeof', EXT('Mapping')) &&
           node.querySelector(CONTENT_SELECTOR)
         ) {
-          const variableInstance = [...node.children]
+          const mappingResource = node.getAttribute('resource');
+
+          if (!mappingResource) {
+            return false;
+          }
+
+          const nodeChildren = [...node.children];
+
+          const type = nodeChildren
+            .find((el) => hasRDFaAttribute(el, 'property', DCT('type')))
+            ?.getAttribute('content');
+
+          if (!type) {
+            return false;
+          }
+
+          const variableInstance = nodeChildren
             .find((el) => hasRDFaAttribute(el, 'property', EXT('instance')))
             ?.getAttribute('resource');
-          const mappingResource = node.getAttribute('resource');
-          const codelistSpan = [...node.children].find((el) =>
+
+          const codelistSpan = nodeChildren.find((el) =>
             hasRDFaAttribute(el, 'property', EXT('codelist'))
           );
+
           const codelistResource =
             codelistSpan?.getAttribute('resource') ??
             codelistSpan?.getAttribute('content');
-          const source = [...node.children]
+
+          const source = nodeChildren
             .find((el) => hasRDFaAttribute(el, 'property', DCT('source')))
             ?.getAttribute('resource');
-          const type = [...node.children]
-            .find((el) => hasRDFaAttribute(el, 'property', DCT('type')))
-            ?.getAttribute('content');
-          const label = node.getAttribute('data-label') || type;
-          const datatype = [...node.children]
+
+          const datatype = nodeChildren
             .find((el) => hasRDFaAttribute(el, 'property', EXT('content')))
             ?.getAttribute('datatype');
-          if (!mappingResource) {
-            return false;
-          }
+
+          const label = node.getAttribute('dataLabel');
```

Comment:
This should be `node.getAttribute('data-label');`. Alternatively, you can also use `node.dataset.label`